pass the value of the android sdk

devoncarew opened the pull request:
- pass the value of the android sdk in `flutter config --machine`
- mark `flutter config` as one of the commands that doesn't need the cache populated (otherwise, querying for the android sdk location could be a very expensive operation)
- pretty print the returned json

@tvolkert, @pq

devoncarew committed:
pass the value of the android sdk

devoncarew commented:
```json
{
  "android-studio-dir": "/Applications/Android Studio 3.0 Preview.app/Contents",
  "android-sdk": "/Users/devoncarew/tools/android-sdk-macosx"
}
```

tvolkert commented on packages/flutter_tools/lib/src/commands/config.dart:
```diff
@@ -39,6 +40,9 @@ class ConfigCommand extends FlutterCommand {
   final List<String> aliases = <String>['configure'];
 
   @override
+  bool get shouldUpdateCache => true;
```
The parent class already defaults to true - is this just to be explicit?

devoncarew replied:
Urk - no. This was a copy-paste error - this should be disabling shouldUpdateCache for `flutter config`.

tvolkert reviewed (commented):
We probably want to allow the user to use `flutter config` to _specify_ the Android SDK directory, for cases where ANDROID_HOME doesn't work (like a dock-launched IDE).

Can the IDE trigger `ConfigCommand` to write preferences?

devoncarew committed:
swap flag

pq reviewed (approved):
LGTM

devoncarew commented:
> We probably want to allow the user to use flutter config to specify the Android SDK directory, for cases where ANDROID_HOME doesn't work (like a dock-launched IDE).

Makes sense - I can add a `flutter config --android-sdk=/foo/bar` option.

> Can the IDE trigger ConfigCommand to write preferences?

Not sure what you mean here. The IDE could pass in `flutter config --android-sdk=/foo/bar`. I don't know that the IDE currently needs to have flutter tools persist the value. Or do you mean that it should write the inferred value (discovered indirectly via the path) to the config file?

tvolkert commented on packages/flutter_tools/lib/src/commands/config.dart:
```diff
@@ -90,8 +94,11 @@ class ConfigCommand extends FlutterCommand {
+    }
 
-    printStatus(JSON.encode(results));
+    printStatus(const JsonEncoder.withIndent('  ').convert(results));
```
IntelliJ can handle formatted JSON?

devoncarew replied:
yup!

tvolkert reviewed (approved):
LGTM

devoncarew committed:
allow the user to set the android-sdk location

tvolkert commented:
> The IDE could pass in `flutter config --android-sdk=/foo/bar`. I don't know that the IDE currently needs to have flutter tools persist the value.

Yeah, that's what I meant.  My thought is that if IntelliJ knows that there's no Android SDK configured as far as flutter tools sees it (which it should be able to know after this PR), then it could allow the user to specify the SDK location in IntelliJ, and that data would get written to the flutter config store.

devoncarew commented:
> My thought is that if IntelliJ knows that there's no Android SDK configured as far as flutter tools sees it (which it should be able to know after this PR), then it could allow the user to specify the SDK location in IntelliJ, and that data would get written to the flutter config store.

Makes sense. Would could implement something like that - the user already has a way in IntelliJ to specify where the android sdk is from the POV of the intellij android tooling.

I updated this PR to allow the user to pass in `flutter config --android-sdk=/foo/bar`, and added tests to match -

tvolkert reviewed (approved):
LGTM!
